deps/spidershim/src/v8trycatch.cc

```diff
@@ -41,7 +41,7 @@ struct TryCatch::Impl {
     // TODO: Propagate exceptions to the caller
     //       https://github.com/mozilla/spidernode/issues/51
   }
-  Isolate* Isolate() const { return isolate_; }
+  Isolate* Iso() const { return isolate_; }
```

Comment:
On Thu, May 05, 2016 at 12:21:27PM -0700, Ehsan Akhgari wrote:
> In deps/spidershim/src/v8trycatch.cc:
> 
> > @@ -41,7 +41,7 @@ struct TryCatch::Impl {
> >      // TODO: Propagate exceptions to the caller
> >      //       https://github.com/mozilla/spidernode/issues/51
> >    }
> > -  Isolate* Isolate() const { return isolate_; }
> > +  Isolate* Iso() const { return isolate_; }
> 
> Do you mind pasting the test case here? This program builds fine with gcc5 for
> me:
> 
> class Foo;
> class Bar {
>   class Foo* Foo();
> };

I was really suprised that works for you, so I poked at it quickly.
Indeed that is fine (I'm not sure what the spec says but it compiles).
However this does not

class Isolate {};
class Foo {
	Isolate* isolate_;
	public:
	class Isolate* Isolate() const;
};

Which makes sense mod my not understanding name lookup, the type of
Isolate_ could be unclear.  However this is fine:

class Isolate {};
class Foo {
	class Isolate* isolate_;
	public:
	class Isolate* Isolate() const;
};

So I guess if you make it clear all over you mean the class it works?
I'm not sure where all you'd need to add class.